### PR TITLE
#166790000 Users Should Be Able To Fetch Ratings Of An Article

### PR DESCRIPTION
**Pull request by kodek-sleuth**
#### What does this PR do?
Enable a user to fetch ratings of an article
#### Description of Task to be completed
Set up a method to fetch ratings of an article
#### How should this be manually tested?
1. Run the following commands in the terminal
```
git clone https://github.com/andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-fetch-article-rating-166790000
npm install
npm start
```
2. Create a .env file and add the following snippets
```
DATABASE_URL=postgres://username:password@localhost:5432/authorshaven
DATABASE_=postgres://username:password@localhost:5432/testing-ah
SECRET=whatever you want it to be
```
3. Install `sequelize` globally
```
sudo npm install -g sequelize
```
4.  Create a database using `sequelize db:create`
5. Install and setup Postman to consume CRUD endpoints
6. Run the following requests in Postman using `localhost:3000`
## Endpoints
|  Method  |  Endpoint  |  Task  |
|  --- |  --- |  ---  |
|  `GET`  |  `/api/article/:articleSlug/ratings`  |  `fetches all article ratings`  |
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166790000](https://www.pivotaltracker.com/story/show/166790000)
#### Screenshots (if appropriate)
N/A
#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by niyoceles — DISMISSED** [review]

**Review by IsaiahRn — DISMISSED** [review]
it can be merged!

**Review by Hervera — CHANGES_REQUESTED** [review]
Remove conflicts & use Sequelize instead of SQL queries

**Review by niyoceles — APPROVED** [review]

**Review by IsaiahRn — DISMISSED** [review]
ready to be merged

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

File: migrations/20190704173059-createTableUsers.js [hunk 1]
```diff
@@ -49,3 +49,4 @@
   });
   
 export const down = (queryInterface, Sequelize) => queryInterface.dropTable('users');
+
```
Comment (Hervera) [inline migrations/20190704173059-createTableUsers.js]: Add the second blank line

File: routes/api/ratings.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import { checkToken } from '../../middlewares';
+import ratingController from '../../controllers/ratings';
+import ratingValidations from '../../middlewares/validations/articleValidations';
+
+const router = express.Router();
+router.post('/article/:articleSlug/rate', checkToken, ratingValidations.validateRating, ratingController.rateArticle);
+router.get('/article/:articleSlug/rating', ratingController.getRating);
```
Comment (Hervera) [inline routes/api/ratings.js]: Make `rating` in plural, to be `ratings` as you are fetching article's ratings

**Review by niyoceles — CHANGES_REQUESTED** [review]
rename `ratings.js` in the controller folder to  `rating.js` all the files in a controller folder must be in single

**Review by Hervera — CHANGES_REQUESTED** [review]
Remove the conflicts

File: controllers/ratings.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      } = findArticle;
+
+
+      const user = await users.findOne({ where: { email: req.decoded.email } });
+
+      const findRate = await ratings.findOne({
+        where: { articleSlug: slug, userId: user.id }, attributes: ['rating']
+      });
+
+      switch (true) {
```
Comment (elemanhillary-zz) [inline controllers/ratings.js]: no `default` case for the switch statement.

File: controllers/ratings.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        case findRate === null:
```
Comment (elemanhillary-zz) [inline controllers/ratings.js]: in which cases are you using the `switch` statement
```javascript
if (!findRate) {
  await ratings.create({
    rating,
    articleSlug: slug,
    userId: user.id
  });
  return res.status(201).json({
    message: 'you have successfully rated this article',
    rating: {
      rating,
      author: author.username,
      title,
      body
    }
  });
}
return res.status(200).json({
  message: 'you already rated this article',
  rating
});
```
may my request be put under your consideration. But kindly compare and check before you change

File: middlewares/checkArticleOwner.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import models from '../models/index';
```
Comment (elemanhillary-zz) [inline middlewares/checkArticleOwner.js]: this file was renamed to `checkResourceOwner`

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
resolve the merge conflicts

**Review by IsaiahRn — DISMISSED** [review]
It can be merged!

**Review by niyoceles — DISMISSED** [review]

**Review by Hervera — DISMISSED** [review]

**Review by IsaiahRn — APPROVED** [review]
It can be merged!

**Review by IsaiahRn — CHANGES_REQUESTED** [review]
solve merge conflicts

**Review by IsaiahRn — DISMISSED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Review by niyoceles — APPROVED** [review]

**Review by Hervera — APPROVED** [review]
Ready to be merged

**Pull request merged**